Dockerfile

```
@@ -0,0 +1,39 @@
+FROM debian:12
+
+# Install dependencies
+RUN apt update -y && apt install -y lsb-release \
+    wget \
+    curl \
+    git \
+    build-essential \
+    libclang-dev \
+    libz-dev
+
+# Install LLVM 16
+RUN echo "deb http://apt.llvm.org/bookworm/ llvm-toolchain-bookworm-16 main" > /etc/apt/sources.list.d/llvm-16.list
+RUN echo "deb-src http://apt.llvm.org/bookworm/ llvm-toolchain-bookworm-16 main" >> /etc/apt/sources.list.d/llvm-16.list
+RUN wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
+RUN apt update -y && apt install -y llvm-16 \
+    libmlir-16-dev \
+    mlir-16-tools \
+    libpolly-16-dev
+
+# Install rust nightly-2023-06-19 (1.72.0-nightly)
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | bash -s -- -y --default-toolchain=nightly-2023-06-19
+ENV PATH="/root/.cargo/bin:${PATH}"
+
+# Copy cairo_native code
+COPY . /cairo_native/
+
+# Fetch cairo corelibs
+RUN git clone --depth 1 \
+    --branch v1.1.0 \
```

the branch needs to be updated
```suggestion
    --branch v2.0.1 \
```

---

Or invoke the script which does that (the one ni `./scripts/fetch-corelibs.sh`)